src/Servers/IIS/IIS/src/Core/IISHttpContextOfT.cs

```diff
@@ -43,6 +44,18 @@ public override async Task<bool> ProcessRequestAsync()
             }
             catch (Exception ex)
             {
+                if (ex is OperationCanceledException && _requestAborted)
+                {
+                    ReportRequestAborted();
+
+                    if (!HasResponseStarted)
+                    {
+                        StatusCode = StatusCodes.Status499ClientClosedRequest;
+                    }
+
+                    return false;
+                }
+
                 ReportApplicationError(ex);
```

Comment:
I'm real hesitant to change the control flow here. This completes a bunch of pipes and stuff like that. I'd stick to just adjusting the logs and changing the status code.

```suggestion
                if (ex is OperationCanceledException && _requestAborted)
                {
                    ReportRequestAborted();
                }
                else
                {
                    ReportApplicationError(ex);
                }
```

To keep the control flow the same, but avoid the status code getting overwritten by 0, you need to modify code below. On line 85, I'd change the following:

```diff
            if (!_requestAborted)
            {
                await ProduceEnd();
            }
            else if (!HasResponseStarted && _requestRejectedException == null)
            {
-                // If the request was aborted and no response was sent, there's no
-                // meaningful status code to log.
-                StatusCode = 0;
+                // If the request was aborted and no response was sent, we use status code 499 for logging
+                // https://learn.microsoft.com/en-us/azure/application-gateway/http-response-codes#499--client-closed-the-connection
+                StatusCode = StatusCodes.Status499ClientClosedRequest;
                success = false;
            }
```

This is the code path that was intended to handle setting the status code of aborted requests. That is if we end up deciding to copy nginx and use 499 instead of 0 for this case like we had previously.